### PR TITLE
Fix APM adjustMavlinkMessage

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
Previous code was horribly wrong.

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
@vooon Can you take a look now?

**Comment by vooon** [comment]
Looks better, but `mavlink_msg_to_send_buffer()` should be after `mavlink_finalize_message_chan()`.
Else there no effect from finalize.

**Comment by DonLakeFlyer** [comment]
Interesting, That's the way it was in the previous implementation as well. I'll change.

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -260,20 +260,17 @@ void Vehicle::_sendMessage(mavlink_message_t message)
         if (link->isConnected()) {
             MAVLinkProtocol* mavlink = MAVLinkProtocol::instance();
             
-            // Write message into buffer, prepending start sign
-            uint8_t buffer[MAVLINK_MAX_PACKET_LEN];
-            int len = mavlink_msg_to_send_buffer(buffer, &message);
+            // Give the plugin a chance to adjust
+            _firmwarePlugin->adjustMavlinkMessage(&message);
+            
             static uint8_t messageKeys[256] = MAVLINK_MESSAGE_CRCS;
```
Comment (vooon) [inline src/Vehicle/Vehicle.cc]: Also perhaps here should be `static const uint8_t`.

Comment (DonLakeFlyer) [inline src/Vehicle/Vehicle.cc]: Sure

**Pull request merged**